classes/class-pmpro-subscription.php

```diff
@@ -1048,47 +1048,59 @@ public function save() {
 	 *
 	 * @since TBD
 	 *
-	 * @return bool True if the subscription was canceled successfully.
+	 * @return bool True if the subscription was canceled successfully in the payment gateway.
 	 */
 	public function cancel() {
-		// Cancel subscription at gateway first.
-		$gateway_object = $this->get_gateway_object();
+		// Prevent infinite loops when calling gateway's cancel() method and passing an order.
+		static $cancelled_subscription_ids = [];
```

Comment:
This makes it difficult to automate tests for. When we get to that point, we may need to move this into a property on the class possibly.